File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(count):
    return sum([number for number in range(count + 1)])**2


def sum_of_squares(count):
    return sum([number**2 for number in range(count +1)])


def difference(count):
    return square_of_sum(count) - sum_of_squares(count)

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(number)
    @number = number
  end

  def square_of_sum
    (0..@number).reduce(:+) ** 2
  end

  def sum_of_squares
    (0..@number).inject(0){ |sum, x| sum + x ** 2 }
  end

  def difference
    square_of_sum - sum_of_squares
  end
end